docs/integrations/engines/snowflake.md

```diff
@@ -0,0 +1,61 @@
+# Snowflake
+
+## Local/Built-in Scheduler
+**Engine Adapter Type**: `snowflake`
+
+### Installation
+```
+pip install "sqlmesh[snowflake]"
+```
+
+### Connection options
+
+| Option          | Description                        |  Type  | Required |
+|-----------------|------------------------------------|:------:|:--------:|
+| `user`          | The Snowflake username             | string |    N     |
+| `password`      | The Snowflake password             | string |    N     |
+| `authenticator` | The Snowflake authenticator method | string |    N     |
+| `account`       | The Snowflake account name         | string |    Y     |
+| `warehouse`     | The Snowflake warehouse name       | string |    N     |
+| `database`      | The Snowflake database name        | string |    N     |
+| `role`          | The Snowflake role name            | string |    N     |
+
+### Snowflake SSO Authorization
+
+SQLMesh supports Snowflake SSO authorization connections using the `externalbrowser` authenticator method. For example:
+
+```yaml
+gateways:
+    snowflake:
+        connection:
+            type: snowflake
+            user: ************
+            authenticator: externalbrowser
+            account: ************
+            warehouse: ************
+            database: ************
+            role: ************
```

Comment:
I think account-role are all optional (and maybe even user) but as a general example this is fine. Sadly I can't test this myself since our account isn't setup with SSO.